File: Python/programmers/etc/level3/연속_펄스_부분_수열의_합.py
def solution(sequence):
    sequence = [0]+[-1*val if idx%2 else val for idx, val in enumerate(sequence)]
    
    for i in range(len(sequence)-1):
        sequence[i+1] += sequence[i]
    
    return abs(max(sequence)-min(sequence))


sequence = [2, 3, -6, 1, 3, -1, 2, 4]
print(solution(sequence)) # 10

File: Java/programmers/etc/level3/연속펄스부분수열의합.java
package CodeTest.Java.programmers.etc.level3;

public class 연속펄스부분수열의합 {
    public long solution(int[] sequence) {
        long[] arr = new long[sequence.length+1];
        
        for (int i=0; i<sequence.length; i++) {
            arr[i+1] = i%2 == 1 ? -sequence[i] : sequence[i];
        }

        for (int i=0; i<arr.length-1; i++) {
            arr[i+1] += arr[i];
        }

        long _min = Long.MAX_VALUE, _max = Long.MIN_VALUE;
        for (int i=0; i<arr.length; i++) {
            _min = Math.min(_min, arr[i]);
            _max = Math.max(_max, arr[i]);
        }

        return _max-_min;
    }
}
